src/app/(sidebar)/transaction/build/page.tsx

```diff
@@ -1,45 +1,350 @@
+      case "memo":
+        return "Memo";
+      default:
+        return "";
+    }
+  };
+
+  const getParamsError = () => {
+    const allErrorMessages: string[] = [];
+    const errors = Object.keys(paramsError);
+
+    // Make sure we don't show multiple errors for the same field
+    const missingParams = missingRequiredParams().filter(
+      (m) => !errors.includes(m),
+    );
+
+    // Missing params
+    if (missingParams.length > 0) {
+      const missingParamsMsg = missingParams.reduce((res, cur) => {
+        return [...res, `${getFieldLabel(cur)} is a required field`];
+      }, [] as string[]);
+
+      allErrorMessages.push(...missingParamsMsg);
+    }
+
+    // Memo value
+    const memoValue = txnParams.memo;
+
+    if (
+      typeof memoValue === "object" &&
+      !isEmptyObject(memoValue) &&
+      !Object.values(memoValue)[0]
+    ) {
+      allErrorMessages.push(
+        "Memo value is required when memo type is selected",
+      );
+    }
+
+    // Fields with errors
+    if (!isEmptyObject(paramsError)) {
+      const fieldErrors = errors.reduce((res, cur) => {
+        return [
+          ...res,
+          `${getFieldLabel(cur as ParamsField)} field has an error`,
+        ];
+      }, [] as string[]);
+
+      allErrorMessages.push(...fieldErrors);
+    }
+
+    return allErrorMessages;
+  };
+
+  const txnJsonToXdr = () => {
+    if (getParamsError().length !== 0) {
+      return {};
+    }
+
+    try {
+      // TODO: remove this formatter once Stellar XDR supports strings for numbers.
+      // Format values to meet XDR requirements
+      const prepTxnParams = Object.entries(txnParams).reduce((res, param) => {
+        const key = param[0] as ParamsField;
+        // Casting to any type for simplicity
+        const value = param[1] as any;
+
+        let val;
+
+        switch (key) {
+          case "seq_num":
+            val = BigInt(value);
+            break;
+          case "fee":
+            val = BigInt(value);
+            break;
+          case "cond":
+            // eslint-disable-next-line no-case-declarations
+            const minTime = value?.time?.min_time;
+            // eslint-disable-next-line no-case-declarations
+            const maxTime = value?.time?.max_time;
+
+            val = {
+              time: {
+                min_time: minTime ? BigInt(minTime) : 0,
+                max_time: maxTime ? BigInt(maxTime) : 0,
+              },
+            };
+            break;
+          case "memo":
+            // eslint-disable-next-line no-case-declarations
+            const memoId = value?.id;
+
+            if (memoId) {
+              val = { id: BigInt(memoId) };
+            } else {
+              val =
+                typeof value === "object" && isEmptyObject(value)
+                  ? "none"
+                  : value;
+            }
+
+            break;
+          default:
+            val = value;
+        }
+
+        return { ...res, [key]: val };
+      }, {});
```

Comment:
This code won't be needed once Stellar XDR supports string numbers.

Reply:
should we add `@TODO` comment

Reply:
:+1: There is a `TODO:` on line 260.